playbooks/library/vault_read.py

```diff
@@ -9,12 +9,15 @@
 class VaultReadModule():
     argument_spec = dict(
         vault_addr=dict(type='str', required=True),
-        role_id=dict(type='str', required=True),
-        secret_id=dict(type='str', no_log=True, required=True),
+        token=dict(type='str', required=False, no_log=True),
+        role_id=dict(type='str', required=False),
+        secret_id=dict(type='str', no_log=True, required=False),
         secret_name=dict(type='str', required=True),
     )
     module_kwargs = {
-        'supports_check_mode': True
+        'supports_check_mode': True,
+        'required_together': ['role_id', 'secret_id'],
```

Comment:
maybe it should be `[('role_id', 'secret_id')],`?

Reply:
good noticed. Thanks